api/ipfsproxy/ipfsproxy_test.go

```diff
@@ -314,6 +314,108 @@ func TestIPFSProxyUnpin(t *testing.T) {
 	}
 }
 
+func TestIPFSProxyPinUpdate(t *testing.T) {
+	ctx := context.Background()
+	proxy, mock := testIPFSProxy(t)
+	defer mock.Close()
+	defer proxy.Shutdown(ctx)
+
+	t.Run("pin/update bad args", func(t *testing.T) {
+		res, err := http.Post(fmt.Sprintf("%s/pin/update", proxyURL(proxy)), "", nil)
+		if err != nil {
+			t.Fatal("request should complete: ", err)
+		}
+
+		defer res.Body.Close()
+		if res.StatusCode != http.StatusInternalServerError {
+			t.Error("request should not be successful with a no arguments")
+		}
+
+		res2, err := http.Post(fmt.Sprintf("%s/pin/update?arg=%s", proxyURL(proxy), test.PathIPFS1), "", nil)
+		if err != nil {
+			t.Fatal("request should complete: ", err)
+		}
+
+		defer res2.Body.Close()
+		if res2.StatusCode != http.StatusInternalServerError {
+			t.Error("request should not be successful with a single argument")
+		}
+	})
+
+	t.Run("pin/update", func(t *testing.T) {
+		res, err := http.Post(fmt.Sprintf("%s/pin/update?arg=%s&arg=%s", proxyURL(proxy), test.PathIPFS1, test.PathIPFS2), "", nil)
+		if err != nil {
+			t.Fatal("request should complete: ", err)
+		}
+
+		defer res.Body.Close()
+
+		var resp ipfsPinOpResp
+		resBytes, _ := ioutil.ReadAll(res.Body)
+		err = json.Unmarshal(resBytes, &resp)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if len(resp.Pins) != 2 ||
+			resp.Pins[0] != test.PathIPFS1 ||
+			resp.Pins[1] != test.PathIPFS2 {
+			t.Errorf("bad response: %s", string(resBytes))
```

Comment:
If `to` is recursively pinned and `from` is not, ipfs throws an error, but our hijacked API succeeds.
Not sure what would be the right thing to here, us to mimick ipfs or otherwise.
```
$ curl "http://localhost:5101/api/v0/pin/update?arg=QmNMgLz2jbCNpsqoBvYzzHZkkTbMbiuAxgjhcztrserJXC&arg=Qmdejgw83ccxAGb2ALDRJCso8YVwkqT992KTr9yhkX4ynk&unpin=false"
{"Message":"'from' cid was not recursively pinned already","Code":0,"Type":"error"}
```

Reply:
Our API fails when from is not pinned (or should, as it is now). There's a test for this.

Reply:
> Our API fails when from is not pinned (or should, as it is now). There's a test for this.

It's not just about being pinned or not. It's about whether it is recursively pinned or not.

Reply:
Yes I checked this. Re-open if I missed something.

Reply:
Strangely enough I can't reproduce it again on ipfs, even after using the same files. So, let it be.